src/pages/concepts/message-bus.mdx

```diff
@@ -0,0 +1,97 @@
+import Wrapper from "../../timvir/wrapper";
+export default Wrapper;
+
+# Message Bus
+
+Components on each Timvir page have access to a page-wide message bus. This bus provides
+a standardized communication channel so that components can communicate with each
+other. The main purpose is to allow implementing micro-interactions that combine multiple
+components on a page.
+
+The bus uses [wonka](https://wonka.kitten.sh/) (a streams/observable library) internally. Components
+can subscribe to all messages that are sent to the bus, optionally filter them to receive
+only those they are interested in.
+
+## API
+
+### useContext
+
+The wonka [Subject](https://wonka.kitten.sh/api/sources#makesubject) is stored in a React Context.
+To access it use the `useContext()` function from `@timvir/core`.
+
+```typescript
+import { useContext } from "@timvir/core";
+
+function Component() {
+  const { bus } = useContext();
+}
+```
+
+Once you have access to the bus, you can subscribe to the messages. The following example
+stores the last message in local React state.
+
+```typescript
+import * as React from "react";
+import { useContext } from "@timvir/core";
+import { pipe, subscribe } from "wonka";
+
+function Component() {
+  const { bus } = useContext();
+
+  const [message, setMessage] = React.useState();
+  React.useEffect(() => pipe(bus.source, subscribe(setMessage)).unsubscribe, [bus]);
+}
+```
+
+### useProps / send
+
+One common use case is to (temporarily) override the props of another component on the page.
+For this purpose we have the `useProps()` hook and `send()` function.
+
+If a component has an `id` (the standard HTML id attribute, a string), then it can be
+targetted via the `send()` function. The following example instructs the component
+with id `some-component-id` to use `rebeccapurple` instead of the original
+value of the `color` prop.
+
+```typescript
+const context = useContext();
+send(context, "some-component-id", "merge", { color: "rebeccapurple" });
```

Comment:
I find the `send` function signature hard to understand. Why not something like this:

```
const {sendMerge} = useTimvirBus()
sendMerge('some-component-id', { color: 'rebeccapurple' });
```

Reply:
I'm considering packaging up the `useProps()` and `send()` functions into a separate module (or even a separate npm package). Naming of most of the API is still open…

```javascript
import { useProps, mergeProps } from "@timvir/bus/props"
```